lib/evse_security.cpp

```diff
@@ -230,14 +237,50 @@ DeleteCertificateResult EvseSecurity::delete_certificate(const CertificateHashDa
     for (const auto& leaf_certificate_path :
          {directories.secc_leaf_cert_directory, directories.csms_leaf_cert_directory}) {
         try {
+            bool secc = (leaf_certificate_path == directories.secc_leaf_cert_directory);
+            bool csms = (leaf_certificate_path == directories.csms_leaf_cert_directory) ||
+                        (directories.csms_leaf_cert_directory == directories.secc_leaf_cert_directory);
+
+            CaCertificateType load;
+
+            if (secc)
+                load = CaCertificateType::V2G;
+            else if (csms)
+                load = CaCertificateType::CSMS;
+
+            // Also load the roots since we need to build the hierarchy for correct certificate hashes
+            X509CertificateBundle root_bundle(ca_bundle_path_map[load], EncodingFormat::PEM);
             X509CertificateBundle leaf_bundle(leaf_certificate_path, EncodingFormat::PEM);
 
```

Comment:
Not a comment but more of a question: Where would sub-CA certificates be found? Is it in the `root_bundle`?
If so, would this method be able to delete sub-CA certificates? The code below only seems to delete from the `leaf_bundle`.